src/providers/tpm_provider/utils.rs

```diff
@@ -84,3 +91,134 @@ pub struct PasswordContext {
     pub context: TpmsContext,
     pub auth_value: Vec<u8>,
 }
+
+pub fn parsec_to_tpm_params(attributes: KeyAttributes) -> Result<KeyParams> {
+    match attributes.key_type {
+        KeyType::RsaKeyPair => {
+            let size = match attributes.key_bits {
+                x @ 1024 | x @ 2048 | x @ 3072 | x @ 4096 => x.try_into().unwrap(), // will not fail on the matched values
+                _ => return Err(ResponseStatus::PsaErrorInvalidArgument),
+            };
+            Ok(KeyParams::Rsa {
+                size,
+                scheme: convert_asym_scheme_to_tpm(attributes.key_policy.key_algorithm)?,
+                pub_exponent: 0,
+            })
+        }
+        KeyType::EccKeyPair { curve_family } => Ok(KeyParams::Ecc {
+            scheme: convert_asym_scheme_to_tpm(attributes.key_policy.key_algorithm)?,
```

Comment:
Does the `scheme` field of `KeyParams::Ecc` carry the same meaning as the algorithm in the PSA policy?
My understanding is that the `scheme` is the algorithm used with the key for sign/verify operation whereas the policy algorithm is the set of algorithms allowed to be used with taht key.
How would you handle a policy algorithm with `Hash::Any` for example? (I am not criticising, just raising general questions!)
Can the `scheme` of a key be changed on-the-fly on an operation per operation level? As in, changing it with the `alg` field of the Parsec `Operation` that have it.

Reply:
> Does the scheme field of KeyParams::Ecc carry the same meaning as the algorithm in the PSA policy?

> How would you handle a policy algorithm with Hash::Any for example?

> Can the scheme of a key be changed on-the-fly on an operation per operation level?

Yes, although in TPM world there isn't any wildcard, so the scheme on the key and the one used when signing/verifying have to be the same. Also, like in the PSA spec, there isn't any way of changing the scheme associated with a key. Hence the method below, `convert_asym_scheme_to_tpm`, does the correct thing by returning `PsaErrorNotSupported` when anything like that is encountered.

Reply:
Ok, that makes sense!
We will need to make sure to verify, test and document that wildcard algorithms are not supported in the policy for TPM keys so that the algorithm in the key policy **is** the algotithm that the key will use.

Reply:
I've added a test to verify that creating a key with the wildcard algorithm returns `NotSupported`. I've also changed the `parsec_to_tpm_signature` method to take in an `AsymmetricSignature` value too, given from the operation object, to use for setting the asymmetric scheme for the signature, it seemed more correct to do it that way. But the method you highlighted here wasn't changed because this is literally about converting the policy used on the key rather than on a particular use of the key.